builders_guide.md

```diff
@@ -0,0 +1,547 @@
+
+Follow the steps in the [README](README.md#static-deployment-of-dark-forest-no-webserver) to deploy the contracts and client.
+
+If your round includes custom planets, run the following command after deploying DarkForestCore:
+
+### 4.1 Add custom planet (optional).
+
+- `yarn workspace eth hardhat:prod game:createPlanets`
+    - This functionality is built-in to the local deploy process, but needs to happen manually for deploying to production.
+
+
+## Side Quest 1: Deploying the subgraph
+
+- What is [The Graph](https://thegraph.com/) and why is it useful?
+    - [The Graph](https://thegraph.com/) is an indexing protocol for querying networks like Ethereum and IPFS. Anyone can build and publish open APIs, called subgraphs, making data easily accessible.
+    - The graph enables more complex queries like who owned this planet at time *t* and has a much faster performance than a vanilla JSON-RPC request.
+- The amazing Jacob Rosenthal has created a DF subgraph that can be deployed via just a few commands. Follow his instructions [here](https://github.com/darkforest-eth/eth/tree/master/subgraph) to deploy the subgraph.
+    - *Note: If you’re on a Mac M1, Docker might not work, and you won’t be able to test the subgraph locally.*
+- If you are changing a fundamental data structure of the subgraph, you will need to update the schema and mappings. See [here](https://github.com/cha0sg0d/eth/commit/1b382624c986ed2b045b8465bde2df543dc5d3f0) for an example edit.
+
+## Side Quest 2: Auto Whitelisting with no server.
+
+*Warning: This information is potentially out of date*
+
+- Get a csv file of all the addresses you want to add to the game.
+- The following command assumes you already have deployed your contracts to production
+- Run:
+    - `yarn workspace eth hardhat:prod whitelist-nokey:register --path <csv of addresses>`
+
+## Side Quest 3: Publishing your new contracts as an npm package.
+
+- Will update more later.
+
+
+## Troubleshooting
+
+- If you see any zk checks fail, like `failed init proof check`, you need to recompile the circuits with the following command:
+    - `yarn workspace eth circom:dev`
+    - Wait ~5 minutes for the circuits to recompile. This will result in a new `Verifier.sol` file.
+- If you see the following warning
+    
+    > Warning: Contract code size is 24622 bytes and exceeds 24576 bytes (a limit introduced in Spurious Dragon). This contract may not be deployable on mainnet. Consider enabling the optimizer (with a low "runs" value!), turning off revert strings, or using libraries.
+    > 
+    - You need to reduce the contract size. Try commenting out changeAdmin or buyHat functions in DarkForestCore. Note that if you do, you will also have to comment out or delete the tests that use these functions.
+
+## Appendix 1: Using Git effectively.
+
+- For setting up your environment initially, see the [Better Method for Running a Local Game](README.md#better-method-for-running-a-local-game)
```

Comment:
Need to update this link, it has the old anchor (and title)

Reply:
will do